### PR TITLE
Backup and restore any user defined grants for extensions

### PR DESCRIPTION
**Pull request by lavjain**
Currently, any grants given on functions that belong to an extension are not being backed up.

This commit allows backing up of any grants that were given on such objects.

e.g.:
```
CREATE EXTENSION IF NOT EXISTS pgcrypto ;
GRANT EXECUTE ON FUNCTION gen_random_bytes(integer) to testrole WITH GRANT OPTION;
```
Result: (added by this commit)
```
REVOKE ALL ON FUNCTION gen_random_bytes(integer) FROM PUBLIC;
REVOKE ALL ON FUNCTION gen_random_bytes(integer) FROM ljain;
GRANT ALL ON FUNCTION gen_random_bytes(integer) TO PUBLIC;
GRANT ALL ON FUNCTION gen_random_bytes(integer) TO testrole WITH GRANT OPTION;
GRANT ALL ON FUNCTION gen_random_bytes(integer) TO ljain;
```
The owner statement is not generated on purpose:
```
ALTER FUNCTION gen_random_bytes(integer OWNER TO ljain;
```

### CODE REVIEW

File: backup/predata_acl.go [hunk 1]
```diff
@@ -87,6 +111,23 @@ func PrintObjectMetadata(file *utils.FileWithByteCount, toc *toc.TOC, metadata O
...
+		if owner := obj.GetOwnerStatement(fqn, entry.ObjectType); owner != "" {
+			statements = append(statements, strings.TrimSpace(owner))
+		}
+		if privileges := obj.GetPrivilegesStatements(fqn, "FUNCTION"); privileges != "" {
+			statements = append(statements, strings.TrimSpace(privileges))
+		}
+		PrintStatements(file, toc, obj, statements)
+	}
+}
+
```
Comment (kyeap-pivotal) [inline backup/predata_acl.go]: Whats the reason for duplicating the function that exists to print ACLs? This look very similar to `PrintObjectMetadata`, which is what is called when functions and aggregates print ACLs.

Comment (kyeap-pivotal) [reply backup/predata_acl.go]: Answered by Lav: to leave out extension fucntion comments.

File: backup/predata_acl.go [hunk 2]
```diff
@@ -54,7 +57,27 @@ type ACL struct {
...
+		}
+}
+
+func (obj ObjectMetadata) FQN() string {
+	if obj.Schema == "" {
+		return obj.Name
+	}
+	return utils.MakeFQN(obj.Schema, obj.Name)
+}
+
```
Comment (kyeap-pivotal) [inline backup/predata_acl.go]: As discussed with @lavjain , we might be able to get rid of this if we check if objects are functions using the funcInfoMap

**Comment by lavjain** [comment]
Rebased with master

**Pull request merged**